crypto/src/pubkey_bytes.rs

```diff
@@ -67,18 +67,27 @@ impl<T, const N: usize> Default for PublicKeyBytes<T, N> {
     }
 }
 
+impl<T: VerifyingKey, const N: usize> FromStr for PublicKeyBytes<T, N> {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let s = s.strip_prefix("0x").unwrap_or(s);
+        let value = hex::decode(s)?;
+        Self::from_bytes(&value[..]).map_err(|_| anyhow::anyhow!("byte deserialization failed"))
+    }
+}
+
 // This guarantees the security of the constructor of a `PublicKeyBytes` instance
 // TODO: replace this clunky sealed marker trait once feature(associated_const_equality) stabilizes
-mod sealed {
+pub mod sealed {
```

Comment:
The whole point is that this module must be private. 
https://rust-lang.github.io/api-guidelines/future-proofing.html#sealed-traits-protect-against-downstream-implementations-c-sealed

Reply:
This still needs to be turned back to private.